contracts/oracle/VoltSystemOracle.sol

```diff
@@ -1,105 +1,161 @@
+import {CoreRefV2} from "../refs/CoreRefV2.sol";
+import {IOracleV2} from "../oracle/IOracleV2.sol";
 import {IVoltSystemOracle} from "./IVoltSystemOracle.sol";
 
 /// @notice contract that receives a fixed interest rate upon construction,
 /// and then linearly interpolates that rate over a 30.42 day period into the VOLT price
 /// after the oracle start time.
-/// Interest can compound annually. Assumption is that this oracle will only be used until
-/// Volt 2.0 ships. Maximum amount of compounding periods on this contract at 2% APR
-/// is 6192 years, which is more than enough for this use case.
+/// Interest compounds once per month.
 /// @author Elliot Friedman
-contract VoltSystemOracle is IVoltSystemOracle {
-    using Decimal for Decimal.D256;
+contract VoltSystemOracle is IVoltSystemOracle, CoreRefV2, IOracleV2 {
+    using SafeCast for *;
 
-    /// ---------- Mutable Variables ----------
+    /// ----------------------------------------
+    /// ---------  Mutable Variables  ----------
+    /// --------- Single Storage Slot ----------
+    /// ----------------------------------------
 
     /// @notice acts as an accumulator for interest earned in previous periods
     /// returns the oracle price from the end of the last period
-    uint256 public oraclePrice;
+    uint200 private _oraclePrice;
 
     /// @notice start time at which point interest will start accruing, and the
     /// current ScalingPriceOracle price will be snapshotted and saved
-    uint256 public periodStartTime;
-
-    /// ---------- Immutable Variables ----------
+    uint40 private _periodStartTime;
 
     /// @notice current amount that oracle price is inflating by monthly in basis points
-    uint256 public immutable monthlyChangeRateBasisPoints;
+    /// cannot be greater than 65,535 basis points per month
+    uint16 private _monthlyChangeRateBasisPoints;
+
+    /// ---------- Immutable Variable ----------
 
     /// @notice the time frame over which all changes in the APR are applied
     /// one month was chosen because this is a temporary oracle
     uint256 public constant TIMEFRAME = 30.42 days;
 
-    /// @param _monthlyChangeRateBasisPoints monthly change rate in the Volt price
-    /// @param _periodStartTime start time at which oracle starts interpolating prices
-    /// @param _oraclePrice starting oracle price
+    /// @param _core reference
+    /// @param startingMonthlyChangeRateBasisPoints monthly change rate in the Volt price
+    /// @param startingPeriodStartTime start time at which oracle starts interpolating prices
+    /// @param startingoraclePrice starting oracle price
     constructor(
-        uint256 _monthlyChangeRateBasisPoints,
-        uint256 _periodStartTime,
-        uint256 _oraclePrice
-    ) {
-        monthlyChangeRateBasisPoints = _monthlyChangeRateBasisPoints;
-        periodStartTime = _periodStartTime;
-        oraclePrice = _oraclePrice;
+        address _core,
+        uint16 startingMonthlyChangeRateBasisPoints,
```

Comment:
Like this - passing in the params into the constructor as the type that they're expected to be. Makes the interface more obvious for integration contracts